src/main/java/org/tugraz/sysds/runtime/instructions/spark/AggregateUnarySPInstruction.java

```diff
@@ -109,28 +121,84 @@ public void processInstruction( ExecutionContext ec ) {
 		{
 			if( _aggtype == SparkAggType.NONE ) {
 				//in case of no block aggregation, we always drop the correction as well as
-				//use a partitioning-preserving mapvalues 
+				//use a partitioning-preserving mapvalues
 				out = out.mapValues(new RDDUAggValueFunction(auop, mc.getRowsPerBlock(), mc.getColsPerBlock()));
 			}
 			else if( _aggtype == SparkAggType.MULTI_BLOCK ) {
 				//in case of multi-block aggregation, we always keep the correction
 				out = out.mapToPair(new RDDUAggFunction(auop, mc.getRowsPerBlock(), mc.getColsPerBlock()));
 				out = RDDAggregateUtils.aggByKeyStable(out, aggop, false);
-	
-				//drop correction after aggregation if required (aggbykey creates 
+
+				//drop correction after aggregation if required (aggbykey creates
 				//partitioning, drop correction via partitioning-preserving mapvalues)
 				if( auop.aggOp.correctionExists )
 					out = out.mapValues( new AggregateDropCorrectionFunction(aggop) );
 			}
-			
+
 			//put output RDD handle into symbol table
-			updateUnaryAggOutputMatrixCharacteristics(sec, auop.indexFn);
-			sec.setRDDHandleForVariable(output.getName(), out);	
+			updateUnaryAggOutputDataCharacteristics(sec, auop.indexFn);
+			sec.setRDDHandleForVariable(output.getName(), out);
 			sec.addLineageRDD(output.getName(), input1.getName());
 		}
 	}
 
-	private static class RDDUAggFunction implements PairFunction<Tuple2<MatrixIndexes, MatrixBlock>, MatrixIndexes, MatrixBlock> 
+	private void processTensorAggregate(ExecutionContext ec) {
+		SparkExecutionContext sec = (SparkExecutionContext)ec;
+		DataCharacteristics dc = sec.getDataCharacteristics(input1.getName());
+
+		//get input
+		JavaPairRDD<TensorIndexes, TensorBlock> in = sec.getBinaryTensorBlockRDDHandleForVariable( input1.getName() );
+		JavaPairRDD<TensorIndexes, TensorBlock> out = in;
+
+		// TODO: filter input blocks for trace
+
+		//execute unary aggregate operation
+		AggregateUnaryOperator auop = (AggregateUnaryOperator)_optr;
+		AggregateOperator aggop = _aop;
+
+		// TODO: aggregate
+		/*
+		//perform aggregation if necessary and put output into symbol table
```

Comment:
Why is this code completely commented? Wasn't this the main goal of this PR and everything serves to make this instruction work? Anyway, I think it's a good idea to split the changes to DataCharacteristics off, so having this in a separate PR is fine but it would be good to make this work before we move on to the next prototype.

Reply:
Of course, that was a mistake on my part, I will fix this ASAP